### PR TITLE
NO-ISSUE: Use `go:embed` for ignition templates

### PR DESCRIPTION
**Pull request by jhernand**
Currently the ignition files are generated from Go templates that are
defined in constants inside the Go code. Those templates also contain
the unit files for the agent and for other systemd units. Changing those
unit files is uncomfortable, because they need to be manually encoded as
a long JSON string. To make those ignition and systemd files easier to
change this patch moves them to external files, using the `embed`
mechanism introduced in Go 1.16.

This patch also introduces to new template functions named
`executeTemplate` and `toJson` that simplify including templates from
other templates. For example, the `agent.service` template is included
from the `discovery.ign` file with this code:

```
"contents": {{ executeTemplate "agent.service" . | toJson }}
```

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

Manually verified that the ignition files included in the generated .iso files are exactly the same.

## Assignees

/cc @
/cc @

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/ignition/templates/discovery.ign [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+      "contents": { "source": "data:,{{.AGENT_MOTD}}" }
+    }{{if .OKDBinaries | not}},
+    {
+      "overwrite": true,
+      "path": "/etc/multipath.conf",
+      "mode": 420,
+      "user": {
+          "name": "root"
+      },
+      "contents": { "source": "data:text/plain;charset=utf-8;base64,ZGVmYXVsdHMgewogICAgdXNlcl9mcmllbmRseV9uYW1lcyB5ZXMKICAgIGZpbmRfbXVsdGlwYXRocyB5ZXMKICAgIGVuYWJsZV9mb3JlaWduICJeJCIKfQpibGFja2xpc3RfZXhjZXB0aW9ucyB7CiAgICBwcm9wZXJ0eSAiKFNDU0lfSURFTlRffElEX1dXTikiCn0KYmxhY2tsaXN0IHsKfQo=" }
```
Comment (omertuc) [inline internal/ignition/templates/discovery.ign:69]: Could we get this one out into a file as well? Could be nice

Comment (jhernand) [reply internal/ignition/templates/discovery.ign:69]: Yes, that makes sense. In order to do that I'd suggest to create a new `toDataUrl` template function. That way the `multipath.conf` file could be included like this:

```
"contents": { "source": "{{ executeTemplate "multipath.conf" . | toDataUrl }}" }
```

I did that in a separate pull request (#4052) to not add noise to this one. If you want I can add the relevant [commit](https://github.com/openshift/assisted-service/pull/4052/commits/408b6547a5f6cb38f53e79dc2802500306dbb03e) to this pull request.

**Comment by omertuc** [comment]
Some of these files are only relevant for pretty obscure scenarios, so I want to run as many tests as we can on this

**Comment by omertuc** [comment]
[Searching](https://prow.ci.openshift.org/?type=periodic&job=*assisted-tpmv2*) prow.ci.openshift.org we can find the periodic (non pre-submit) variant of this job and look at its [history](https://prow.ci.openshift.org/job-history/gs/origin-ci-test/logs/periodic-ci-openshift-assisted-test-infra-master-e2e-metal-assisted-tpmv2-periodic).

We can see it's failing for the same reason, so very probably safe to ignore

**Comment by osherdp** [comment]
> @osherdp @nmagnezi The "edge-e2e-metal-assisted-tmpv2" test seems to be failing to enable the virtual TPM in one of the nodes of the cluster:
> 
> ```
> Need read rights on issuer certificate /var/lib/swtpm-localca/issuercert.pem for user tss.
> swtpm_localca exit with status 1: 
> An error occurred. Authoring the TPM state failed.
> ```
> 
> See the details [here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_assisted-service/4044/pull-ci-openshift-assisted-service-master-edge-e2e-metal-assisted-tpmv2/1542802573495373824/artifacts/e2e-metal-assisted-tpmv2/assisted-baremetal-gather/artifacts/libvirt-qemu/test-infra-cluster-0edb367a-master-1-swtpm.log).
> 
> I believe you are familiar with that [issue](https://github.com/stefanberger/swtpm/issues/644). Do you know if it has been addressed?
> 
> @omertuc I think that failure is not related to this patch.

yes, the problem is that Rocky Linux that we use (and RHEL 8 in general) has a version of swtpm package that still contains this bug. We're trying to address this, but in the meantime feel free to retest:
/test edge-e2e-metal-assisted-tpmv2

**Review by avishayt — APPROVED** [review]
Great improvement - thanks!

**Pull request merged**